#!/usr/bin/env python3

PACKAGE = "offboard_py"  # Replace with your actual package name

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Trajectory parameters without enum
gen.add("trajectory_type", str_t, 0, "Type of Trajectory", "circle")

gen.add("radius", double_t, 0, "Radius for Circular Trajectory (m)", 40.0, 0.0, 100.0)
gen.add("vertical_amplitude", double_t, 0, "Amplitude for Vertical Oscillation (m)", 1.0, 0.0, 10.0)
gen.add("v", double_t, 0, "Initial Speed (m/s)", 8.0, 0.0, 20.0)

# Add other parameters as needed

exit(gen.generate(PACKAGE, "offboard_py", "TargetConfig"))
